### ISSUE TITLE
How can I repeat the generator?

### ISSUE THREAD

**Issue by airplane2230** [tensorflow/tensorflow#36345]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Window 10
- TensorFlow version (use command below): 2.1
- Python version: 3.7.0

This is my custom generator code as follows:

```import tensorflow as tf
from tensorflow.keras.preprocessing.image import ImageDataGenerator

class DataGenerator(tf.keras.utils.Sequence):
    def __init__(self, df, batch_size = 32, target_size = (112, 112), shuffle = True):
        self.len_df = len(df)
        self.batch_size = batch_size
        self.target_size = target_size
        self.shuffle = shuffle
        self.class_col = ['black', 'blue', 'brown', 'green', 'red', 'white', 
             'dress', 'shirt', 'pants', 'shorts', 'shoes']
        self.generator = ImageDataGenerator(rescale = 1./255)
        self.df_generator = self.generator.flow_from_dataframe(dataframe=df, 
                                                          directory='',
                                                            x_col = 'image',
                                                            y_col = self.class_col,
                                                            target_size = self.target_size,
                                                            color_mode='rgb',
                                                            class_mode='other',
                                                            batch_size=self.batch_size,
                                                            seed=42)
        self.colors_df = df['color']
        self.on_epoch_end()
      
    def __len__(self):
        return int(np.floor(self.len_df) / self.batch_size)
    
    def on_epoch_end(self):
        self.indexes = np.arange(self.len_df)
        if self.shuffle:
            np.random.shuffle(self.indexes)
        
    def __getitem__(self, index):
        indexes = self.indexes[index * self.batch_size : (index + 1) * self.batch_size]
        colors = self.__data_generation(indexes)
        
        images, labels = self.df_generator.__getitem__(index)
        
        # return multi-input and output
        return [images, colors], labels
    
    def __data_generation(self, indexes):
        colors = self.colors_df[indexes]
        
        return colors
```

But, it doesn't repeat in fit() with this warning message!
![image](https://user-images.githubusercontent.com/33315343/73481608-836dca80-43df-11ea-8e5a-00e01e82cd87.png)

**Comment by airplane2230** [comment]
It removes the A and B args and works.
After removing the 'steps_per_epoch' and 'validation_steps', it works.

But there is problem the whole data is not use.

My train data num: 5578
After removing arguments, it uses only 5568.

How can I solve this problem?
Is there something wrong with my custom code?

**Comment by rmothukuru** [comment]
In order to expedite the trouble-shooting process, please provide a code snippet to reproduce the issue reported here. Thanks!

**Comment by tensorflowbutler** [comment]
It has been 20 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 36 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by airplane2230** [comment]
Sorry, Thank you for your check this issue.

The full code is as follows:
[https://drive.google.com/open?id=1stCgY9E3JG0tzT9bC5oNibcXcmE_8SKd](https://drive.google.com/open?id=1stCgY9E3JG0tzT9bC5oNibcXcmE_8SKd)

**Comment by airplane2230** [comment]
+ mistake: close --> reopen

Sorry, Thank you for your check this issue.

The full code is as follows:
[https://drive.google.com/open?id=1stCgY9E3JG0tzT9bC5oNibcXcmE_8SKd](https://drive.google.com/open?id=1stCgY9E3JG0tzT9bC5oNibcXcmE_8SKd)

I haven't solved this problem yet.
Unlike before, I am ready to answer quickly.

**Comment by rmothukuru** [comment]
This question is better asked on [StackOverflow](http://stackoverflow.com/questions/tagged//tensorflow) since it is not a bug or feature request. There is also a larger community that reads questions there.

**Comment by airplane2230** [comment]
I will try it.

Thank you and close this issue.
